src/v/cluster_link/link.cc

```diff
@@ -12,31 +12,200 @@
 #include "cluster_link/link.h"
 
 #include "cluster_link/logger.h"
+#include "model/namespace.h"
 
 namespace cluster_link {
 
-link::link(model::metadata config)
-  : _config(std::move(config)) {}
+using kafka::data::rpc::partition_leader_cache;
+using kafka::data::rpc::partition_manager;
+
+link::link(
+  ::model::node_id self,
+  model::metadata config,
+  partition_leader_cache* partition_leader_cache,
+  partition_manager* partition_manager)
+  : _self(self)
+  , _config(std::move(config))
+  , _partition_leader_cache(partition_leader_cache)
+  , _partition_manager(partition_manager) {}
 
 ss::future<> link::start() {
-    vlog(cllog.info, "Starting panda link {} ({})", _config.name, _config.uuid);
-    return ss::now();
+    vlog(
+      cllog.info, "Starting cluster link {} ({})", _config.name, _config.uuid);
+    for (auto& t : _tasks) {
+        if (should_start_task(t)) {
+            vlog(
+              cllog.info,
+              "Starting task {} for cluster link {} ({})",
+              t->name(),
+              _config.name,
+              _config.uuid);
+            auto res = co_await t->start();
+            if (!res) {
+                vlog(
+                  cllog.error,
+                  "Failed to start task {}: {}",
+                  t->name(),
+                  res.assume_error().message());
+            }
+        } else {
+            vlog(
+              cllog.debug,
+              "Skipping task {} for cluster link {} ({})",
+              t->name(),
+              _config.name,
+              _config.uuid);
+        }
+    }
+    _is_running = true;
 }
 
 ss::future<> link::stop() {
-    vlog(cllog.info, "Stopping panda link {} ({})", _config.name, _config.uuid);
-    return ss::now();
+    vlog(
+      cllog.info, "Stopping cluster link {} ({})", _config.name, _config.uuid);
+    _is_running = false;
+    for (auto& t : _tasks) {
+        vlog(
+          cllog.info,
+          "Stopping task {} for cluster link {}",
+          t->name(),
+          _config.name);
+        auto res = co_await t->stop();
+        if (!res) {
+            vlog(
+              cllog.error,
+              "Failed to stop task {}: {}",
+              t->name(),
+              res.assume_error().message());
+        }
+    }
+}
+
+ss::future<> link::register_task(std::unique_ptr<task> t) {
+    vlog(
+      cllog.debug,
+      "Registering task {} for cluster link {} ({})",
+      t->name(),
+      _config.name,
+      _config.uuid);
+    auto& emplaced_task = _tasks.emplace_back(std::move(t));
+    if (_is_running && should_start_task(emplaced_task)) {
+        vlog(cllog.info, "Starting task {}", emplaced_task->name());
+        auto res = co_await emplaced_task->start();
+        if (!res) {
+            vlog(
+              cllog.error,
+              "Failed to start task {}: {}",
+              emplaced_task->name(),
+              res.assume_error().message());
+        }
+    }
```

Comment:
This is similar to the previous comment, there is completely no error propagation to the caller